From 527b5c34d8504d289e23a3290f02ae210370632b Mon Sep 17 00:00:00 2001
From: Mark Abraham <mark.j.abraham@gmail.com>
Date: Tue, 28 Jan 2025 12:04:47 +0100
Subject: [PATCH] Include cstdio where needed

---
 src/gromacs/mdrun/logging.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gromacs/mdrun/logging.h b/src/gromacs/mdrun/logging.h
index 39557d09a2..78a1f84977 100644
--- a/src/gromacs/mdrun/logging.h
+++ b/src/gromacs/mdrun/logging.h
@@ -43,6 +43,8 @@
 #ifndef GMX_MDRUN_LOGGING_H
 #define GMX_MDRUN_LOGGING_H
 
+#include <cstdio>
+
 #include <memory>
 
 #include "gromacs/utility/basedefinitions.h"
-- 
2.34.1

